kani-compiler/src/codegen_cprover_gotoc/codegen/place.rs

```diff
@@ -365,18 +365,20 @@ impl<'tcx> GotocCtx<'tcx> {
     /// a named variable.
     ///
     /// Recursively finds the actual FnDef from a pointer or box.
-    fn codegen_local_fndef(&mut self, ty: ty::Ty<'tcx>) -> Option<Expr> {
+    fn codegen_local_fndef(&mut self, ty: Ty) -> Option<Expr> {
         match ty.kind() {
             // A local that is itself a FnDef, like Fn::call_once
-            ty::FnDef(defid, args) => Some(self.codegen_fndef(*defid, args, None)),
+            TyKind::RigidTy(RigidTy::FnDef(def, args)) => {
+                Some(self.codegen_fndef_stable(def, &args, None))
+            }
             // A local can be pointer to a FnDef, like Fn::call and Fn::call_mut
-            ty::RawPtr(inner) => self
-                .codegen_local_fndef(inner.ty)
+            TyKind::RigidTy(RigidTy::RawPtr(inner, _)) => self
+                .codegen_local_fndef(inner)
                 .map(|f| if f.can_take_address_of() { f.address_of() } else { f }),
             // A local can be a boxed function pointer
-            ty::Adt(def, _) if def.is_box() => {
-                let boxed_ty = self.codegen_ty(ty);
-                self.codegen_local_fndef(ty.boxed_ty())
+            TyKind::RigidTy(RigidTy::Adt(def, args)) if def.is_box() => {
+                let boxed_ty = self.codegen_ty_stable(ty);
+                self.codegen_local_fndef(*args.0[0].ty().unwrap())
```

Comment:
I don't know what `*args.0[0].ty().unwrap()` is... can we add an intermediate step?

Reply:
ok

Reply:
Yeah, even I had to spend 5min remembering what this meant. I also added a comment to help. LOL